src/main/java/net/fabricmc/loom/providers/MappingsProvider.java

```diff
@@ -66,50 +112,133 @@ public void provide(DependencyInfo dependency, Project project, LoomGradleExtens
 		project.getLogger().lifecycle(":setting up mappings (" + dependency.getDependency().getName() + " " + dependency.getResolvedVersion() + ")");
 
 		String version = dependency.getResolvedVersion();
-		File mappingsJar = dependency.resolveFile().orElseThrow(() -> new RuntimeException("Could not find dependency " + dependency));
+		File mappingsJar = dependency.resolveFile().orElseThrow(() -> new RuntimeException("Could not find yarn mappings: " + dependency));
+
+		// Hacky but seems like the easiest way to get the classifier
+		this.isV2 = mappingsJar.getName().endsWith("-v2.jar");
```

Comment:
Humm, opening up the file might be needed here. Depending on the filename is bad imo.

Reply:
Well we are guaranteed by the way the maven publish plugin works that it postfixes with `-<classifier>`.

Reply:
Some people may well want to use custom mappings that didnt come from that.

Reply:
I think I solved it well